### ISSUE TITLE
Add tt-metal support for torch.argmax, torch.argmin

### ISSUE THREAD

**Issue by farbabi** [tenstorrent-metal/tt-metal#4215]
In Faclon7b demo post-processing, the tt_logits need to be moved to cpu to return the indices of the maximum value of all elements in the logits tensor to produce prefill and decode ids.
Currently, we are missing it in metal. Moving ids to cpu to use torch to generate ids makes the process significantly slow. It is not included when reporting inference and compile time, but the end-to-end speed is lower than the prefill + decode inference time, which is not suitable for customer facing team. Therefore, it is crucial to have tt op version of `torch.argmax` [https://pytorch.org/docs/stable/generated/torch.argmax.html].

Expanding scope of this issue to add argmin as well, since its requested by our partner, since it is algebraically related to argmin as follows,
```
import torch
torch.tensor([1,2,3,4])
t=torch.tensor([1,2,3,4])
u=-t
assert t.argmax() == u.argmin()
```

**Comment by umadevimcw** [comment]
argmax and argmin PR merged to main
supports dimension N, C, H and W, all (if all param is enabled, given dimension will be ignored and return indices across all dimension)
The current implementation return correct indices value if the index range is between 0 - 256 (if the range exceeds) because of this issue https://github.com/tenstorrent-metal/tt-metal/issues/4787  there will be differences in index values. Bfloat16 conversion needs to be fixed for the range > 256
